lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1559,17 +1559,26 @@ sub _failure_reason {
 
     my @failed_modules;
     my $modules = $self->modules;
+
     while (my $m = $modules->next) {
         if ($m->result eq FAILED || $m->result eq SOFTFAILED) {
-            push(@failed_modules, $m->name . ':' . $m->result);
+            # Look for serial failures which have bug reference
+            my @bugrefs = map { find_bugref($_->{title}) || '' } @{$m->details};
+            # If bug reference is in title, put it as a failure reason, otherwise use module name
+            if (my $failure_reason = join('', @bugrefs)) {
+                return $failure_reason;
+            }
+            else {
+                push(@failed_modules, $m->name . ':' . $m->result);
```

Comment:
Hm, can you think of a case where we would find a bugref above but actually want this else-branch to be active? E.g. Can we still override a failed job with a a bugref that a human reviewer specifies for carryover?

Reply:
Yes, that's the whole point for this approach, as it's just failure detection, but we don't change/add the comment, it still has to be given by the reviewer. I don't see any scenarios except these use-cases with kernel bugs where this feature is useful, but we already have 2 of them (one for 42.3 leap upgrade test and second one for sle on arm).